net/src/sim2h_worker.rs

```diff
@@ -0,0 +1,407 @@
+                ))
+            }
+            Lib3hClientProtocol::HandleGetGossipingEntryListResult(entry_list_data) => {
+                //let log_context = "ClientToLib3h::HandleGetGossipingEntryListResult";
+                self.send_wire_message(WireMessage::Lib3hToClientResponse(
+                    Lib3hToClientResponse::HandleGetGossipingEntryListResult(entry_list_data)
+                ))
+            }
+
+            // -- N3h specific functinonality -- //
+            Lib3hClientProtocol::Shutdown => {
+                debug!("Got Lib3hClientProtocol::Shutdown from core in sim2h worker");
+                Ok(())
+            }
+        }
+    }
+
+    fn handle_server_message(&mut self, message: WireMessage) -> NetResult<()> {
+        match message {
+            WireMessage::Ping => self.send_wire_message(WireMessage::Pong)?,
+            WireMessage::Pong => {},
+            WireMessage::Lib3hToClient(m) =>
+                self.to_core.push(Lib3hServerProtocol::from(m)),
+            WireMessage::ClientToLib3hResponse(m) =>
+                self.to_core.push(Lib3hServerProtocol::from(m)),
+            WireMessage::Lib3hToClientResponse(m) =>
+                error!("Got a Lib3hToClientResponse from the Sim2h server, weird! Ignoring: {:?}", m),
+            WireMessage::ClientToLib3h(m) =>
+                error!("Got a ClientToLib3h from the Sim2h server, weird! Ignoring: {:?}", m),
+            WireMessage::Err(sim2h_error) => match sim2h_error {
+                WireError::MessageWhileInLimbo => if let Some(space_data) = self.space_data.clone() {
+                    self.send_wire_message(WireMessage::ClientToLib3h(
+                        ClientToLib3h::JoinSpace(space_data)
+                    ))?;
+                } else {
+                    error!("Uh oh, we got a MessageWhileInLimbo errro and we don't have space data. Did core send a message before sending a join? This should not happen.");
+                }
+                WireError::Other(e) => error!("Got error from Sim2h server: {:?}", e),
+            }
+        };
+        Ok(())
+    }
+}
+
+impl NetWorker for Sim2hWorker {
+    /// We got a message from core
+    /// -> forward it to the NetworkEngine
+    fn receive(&mut self, data: Lib3hClientProtocol) -> NetResult<()> {
+        self.inbox.push(data);
+        Ok(())
+    }
+
+    /// Check for messages from our NetworkEngine
+    fn tick(&mut self) -> NetResult<bool> {
+        self.num_ticks += 1;
+        //if self.num_ticks % 100 == 0 {
+        //    io::stdout().flush()?;
+        //}
+        if let Err(transport_error) = detach_run!(&mut self.transport, |t| t.process(self)) {
+            error!("Transport error: {:?}", transport_error);
+            // This most likely means we have connection issues.
+            // Send ping to reestablish a potentially lost connection.
+            let _ = self.send_wire_message(WireMessage::Ping);
```

Comment:
not a big fan of ignoring results in this manner... It hides problems when debugging.

```suggestion
            if let Err(e) = self.send_wire_message(WireMessage::Ping) {
                debug!("send ping failure: {:?}", e);
            }
```